### DISCUSSION TITLE
Welcome to anker-solix-api Discussions!

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/anker-solix-api#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by khfiebig** [comment]
Hello,
In the version 1.0 I can see my Solix C1000 Device, now, with 1.5.2, I got this error,
in both cases I use the test_api script, can you help or fix the problem ?
thank you, best regards

==== Error version 1.5.2 ====
<class 'AttributeError'>: A1761
Traceback (most recent call last):
  File "/home/anker-solix-api/./test_api.py", line 267, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/anker-solix-api/./test_api.py", line 248, in main
    await myapi.update_device_details()
  File "/home/anker-solix-api/api/api.py", line 1006, in update_device_details
    await self.get_bind_devices(fromFile=fromFile)
  File "/home/anker-solix-api/api/api.py", line 1136, in get_bind_devices
    self._update_dev(device)
  File "/home/anker-solix-api/api/api.py", line 584, in _update_dev
    if getattr(SolixDeviceCapacity, device.get("device_pn", "")):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/enum.py", line 789, in __getattr__
    raise AttributeError(name) from None
AttributeError: A1761

==== Result version 1.0 without error ====
Device Overview:
{
  "AZ19": {
    "device_sn": "AZ19",
    "type": "solarbank",
    "site_id": 0d0",
    "is_admin": true,
    "device_pn": "A17C0",
    "name": "Solarbank E1600",
    "battery_soc": "12",
    "charging_power": "121",
    "power_unit": "W",
    "charging_status": "3",
    "status_description": "unknown",
    "input_power": "211",
    "output_power": "121",
    "set_output_power": "",
    "bt_ble_mac": "E878",
    "wifi_online": true,
    "charge": false,
    "bws_surplus": "0",
    "sw_version": "v1.5.6",
    "auto_upgrade": false,
    "power_cutoff": 10
  },
  "AP00020": {
    "device_sn": "A6N300",
    "device_pn": "A1761",
    "bt_ble_mac": "E8484",
    "name": "SOLIX C1000(X)",
    "wifi_online": true,
    "charge": false,
    "bws_surplus": "0",
    "sw_version": "v1.3.2"
  }
}

**Reply by khfiebig** [reply]
with verison 1.5.5 it works, thank you

**Reply by thomluther** [reply]
Hi, you are the first one with such a device.
It would help if you can run a configuration export to provide randomized sample data how the Api responses look like for such a device or power system you defined for it.
I assume it might have completely different structure and fields and therefore is not really supported with the Api library per default.

**Comment by khfiebig** [comment]
Hi, here ist the configuration export file, I hope it helps
[exportSys.tgz](https://github.com/thomluther/anker-solix-api/files/14560117/exportSys.tgz)

**Comment by thomluther** [comment]
Hi @khfiebig 
Thanks for the export. Unfortunatly there is little to No information from the cloud about your SOLIX C1000.
I wonder whether this is configured completely in your App?
From how I understand the site rules output, the SOLIX C1000 PN A1761 cannot be configured into the same system (site) as your Solarbank. Solarbank systems typically are site type 2, which allows only following PNs for Solarbanks and Inverters.
The bind devices output is also showing that it has no Wifi connection, but only BT?
```
      {
        "device_sn": "1OVQCE1LTA8QITV5",
        "product_code": "A1761",
        "bt_ble_id": "95:F7:0F:1F:CE:E2",
        "bt_ble_mac": "95F70F1FCEE2",
        "device_name": "SOLIX C1000(X)",
        "alias_name": "SOLIX C1000(X)",
        "img_url": "https://public-aiot-fra-prod.s3.dualstack.eu-central-1.amazonaws.com/anker-power/public/product/2024/01/17/iot-admin/b4VVnVyVA7KkQTQR/picl_A1761_normal.png",
        "link_time": 1709980538,
        "wifi_online": false,
        "wifi_name": "",
        "relate_type": [
          "ble"
        ],
        "charge": false,
        "bws_surplus": 0,
        "device_sw_version": "v1.3.2",
        "has_manual": true,
        "hes_data": null
      },

```
So I wonder which capabilities you have in the Anker App to manage the C1000?
Can you create another System and add it to the new system?
Will that give you more management capabilitities and monitoring for the device?
I can also extend the Api with more metrics about the device. Looking at the product page https://www.anker.com/eu-de/products/a1761 it says the capacity is 1056Wh, but expandable to 2112Wh....Would that then be another 'device' for the battery expansion pack that is also managed by the Anker App (Cloud)?
The product page also says 'Bequeme Steuerung per App', so I wonder how you can do that if it is not configured into a System yet.
Your site list is only showing a single site (=system):

**Reply by khfiebig** [reply]
Yes, you are right, at the moment only the the device A1771 is supported to include it in a system.
I opend a feature request by anker, to support the new device A1761 also. but at the moment the support is very bad.
Yes I can manage the device in the app, it's a single device and I have all posibilities, see the screenshots.
![Screenshot_2024-03-12-11-49-57-57_1c5ec63d785335787a5c4d8741dcd510](https://github.com/thomluther/anker-solix-api/assets/150910988/94f896bb-2e00-4d81-930a-1883b53ae5ed)
![Screenshot_2024-03-12-11-51-42-51_1c5ec63d785335787a5c4d8741dcd510](https://github.com/thomluther/anker-solix-api/assets/150910988/9430bdf2-998a-4eb2-b500-f06d6725a337)
![Screenshot_2024-03-12-11-50-37-75_1c5ec63d785335787a5c4d8741dcd510](https://github.com/thomluther/anker-solix-api/assets/150910988/6decbfc8-c31d-4570-80ef-609b5dc08c4c)

**Comment by thomluther** [comment]
Ah ok, so that is management on device level, which is probably only via BT connection. 
Its not going over the cloud I think, therefore I don't see anything in the Api responses.
Can you create a new 2nd System where you only add the C1000 device or is that device number not even supported yet for new Systems?
It is true that I cannot see the A1761 PN in any of the Power Site rules, so maybe it cannot even be added to a new System yet, therefore no power data will be available via Cloud or Api.

**Reply by khfiebig** [reply]
- WLAN is also possible, I do the export again with WLAN enabled
- on the last screenshot you can see at the top the wlan icon
- 
[exportSystem.tgz](https://github.com/thomluther/anker-solix-api/files/14571961/exportSystem.tgz)
-No, I cannot create a system with solix C1000, only a single device is possible
